shapely/geometry/base.py

```diff
@@ -653,6 +656,7 @@ def normalize(self):
         """
         # self.impl['normalize'](self)
         if self._geom is None:
+            # Do we consider this a type error?
```

Comment:
This one we can leave as is. normalize() is part of the public API, yes. But this error will almost never be seen and exists to prevent a null C pointer from being passed to the GEOS function underneath. In Shapely 2.0 the implementation here will change quite a bit and null pointers will be less of a problem, or at least a different problem.